analyzer/protocol/openvpn/main.zeek

```diff
@@ -0,0 +1,90 @@
+module OpenVPN;
+
+export {
+	## Set to true to disable the analyzer after the handshake is detected.
+	## This helps reduce processing if you will not look at all of the OpenVPN
+	## traffic.
+	option disable_analyzer_after_detection = T;
+
+	type ControlMsg: record {
+		## Opcode
+		opcode					:	count;
+		## Key ID
+		key_id					:	count;
+		## Session ID
+		session_id	    		: 	string &optional;
+		## Packet id ack array
+		packet_id_ack_array		:	vector of count &optional;
+		## Remote session ID
+		remote_session_id		:	string &optional;
+		## Packet ID
+		packet_id				:	count &optional;
+		## The amount of data
+		data_len				:	count;
+	};
+
+	type AckMsg: record {
+		## Opcode
+		opcode					:	count;
+		## Key ID
+		key_id					:	count;
+		## Session ID
+		session_id	    		: 	string &optional;
+		## Packet id ack array
+		packet_id_ack_array		:	vector of count &optional;
+		## Remote session ID
+		remote_session_id		:	string &optional;
+	};
+
+	type DataMsg: record {
+		## Opcode
+		opcode					:	count;
+		## Key ID
+		key_id					:	count;
+		## The amount of data
+		data_len				:	count;
+		## Peer ID
+		peer_id		    		: 	string &optional;
+	};
+
+	global OpenVPN::control_message: event(c: connection, is_orig: bool, msg: OpenVPN::ControlMsg);
```

Comment:
It would be nice to document these events with in the typical Zeek documentation style.